### PR TITLE
fix: [Core] update escrow function permissions

### LINKED ISSUE #558: Check the permission of core contract functions.

### PR DESCRIPTION
**Pull request by leric7**
## Description

Escrow functions `bulkPayout`, `storeResult`, and `complete` needs more restricted permission.
And reputation/recording oracle should not be considered super user, in general.

## Summary of changes

- Added two more modifiers. `trustedOrReputationOracle`, `trustedOrRecordingOracle`.
- Changed modifier of three functions `bulkPayout`, `storeResult`, and `complete` of escrow contract.
- Removed `recordingOracle`, and `reputationOracle` from trusted handlers list.

## Related issues
[Keywords for linking issues](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests)

Closes #558 

## Operational checklist

- [ ] All new functionality is covered by tests
- [ ] Any related documentation has been changed or added

### CODE REVIEW

File: packages/core/contracts/Escrow.sol [hunk 1]
```diff
@@ -47,6 +47,9 @@ contract Escrow is IEscrow, ReentrancyGuard {
     string public manifestUrl;
     string public manifestHash;
 
+    string public intermediateResultsUrl;
+    string public intermediateResultsHash;
```
Comment (portuu3) [inline packages/core/contracts/Escrow.sol:51]: I wonder if hash of the files is actually necessary, what do you think?

File: packages/core/contracts/Escrow.sol [hunk 2]
```diff
@@ -297,20 +307,27 @@ contract Escrow is IEscrow, ReentrancyGuard {
...
 
+    modifier trustedOrReputationOracle() {
+        require(
+            areTrustedHandlers[msg.sender] || msg.sender == reputationOracle,
+            'Address calling not trusted'
+        );
+        _;
+    }
+
+    modifier trustedOrRecordingOracle() {
```
Comment (portuu3) [inline packages/core/contracts/Escrow.sol:323]: This modifier might not be worth since it is used only once. Check the weight of the contract, if it is lighter by removing it, we should consider to remove it

File: packages/examples/fortune/reputation-oracle/src/services/escrow.test.ts
Comment (portuu3) [inline packages/examples/fortune/reputation-oracle/src/services/escrow.test.ts:1]: why this change?

**Review by portuu3 — CHANGES_REQUESTED** [review]
We should update sdk and add functions if needed, like getIntermediateResultsUrl (also update docs)

File: packages/examples/fortune/reputation-oracle/src/services/escrow.test.ts
Comment (leric7) [reply packages/examples/fortune/reputation-oracle/src/services/escrow.test.ts:1]: There is a `bulkPayout` function call, and recording oracle now can't call this function.

**Comment by leric7** [comment]
@portuu3 , I resolved your comments, and added sdk functions. I don't know what you meant by updating doc, but if you were referring to updating doc for core contracts, I have another ticket for that. #560

File: packages/core/contracts/Escrow.sol (hunk 1 above)
Comment (leric7) [reply packages/core/contracts/Escrow.sol]: Agreed. Dropped `intermediateResultsHash` param. However, want to keep both function params of `storeResults` for backwards compatibility.

**Comment by leric7** [comment]
> > @portuu3 , I resolved your comments, and added sdk functions. I don't know what you meant by updating doc, but if you were referring to updating doc for core contracts, I have another ticket for that. #560
> 
> I meant this doc https://www.notion.so/human-protocol/SDK-d312448eb4224d1886435e27897189e3

I updated.

**Review by portuu3 — APPROVED** [review]
